asserts/checkers.go

```diff
@@ -38,12 +39,18 @@ func checkMandatory(headers map[string]string, name string) (string, error) {
 	return value, nil
 }
 
-func checkAssertType(assertType AssertionType) (*assertionTypeRegistration, error) {
-	reg := typeRegistry[assertType]
-	if reg == nil {
-		return nil, fmt.Errorf("unknown assertion type: %v", assertType)
+var errInvalidAssertionType = errors.New("invalid assertion type")
```

Comment:
Yeah, sorry, I meant the value is never used. Guess you are just saving the double return. I'd show a more detailed error, and include the reorganization suggested below.